### ISSUE TITLE
Lint to warn on a suspicious unnecessary operation, that happens to have the same type as the return type

### ISSUE THREAD

**Issue by Centri3** [rust-lang/rust-clippy#10941]
### What it does

if there's `ControlFlow::Break(());`, and it's completely unnecessary, whilst the function returns `ControlFlow`, rather than just emitting `no_effect`, clippy should also say that it's likely intended to be returned.

This could be an enhancement to `no_effect` rather than a new lint.

### Advantage

- Saves the developer precious sanity

### Drawbacks

- None.

### Example

```rust
fn yep() -> bool {
	true;
}
```

Should be written as:

```rust
fn yep() -> bool {
	return true;
}
```
(Minus the `return` if it's unnecessary, but other lints will take care of that)
